src/types/dict.rs

```diff
@@ -519,16 +534,23 @@ pub(crate) use borrowed_iter::BorrowedDictIter;
 pub trait IntoPyDict: Sized {
     /// Converts self into a `PyDict` object pointer. Whether pointer owned or borrowed
     /// depends on implementation.
-    fn into_py_dict_bound(self, py: Python<'_>) -> Bound<'_, PyDict>;
+    fn into_py_dict(self, py: Python<'_>) -> Bound<'_, PyDict>;
+
+    /// Deprecated name for [`IntoPyDict::into_py_dict`].
+    #[deprecated(since = "0.23.0", note = "renamed to `IntoPyDict::into_py_dict`")]
+    #[inline]
+    fn into_py_dict_bound(self, py: Python<'_>) -> Bound<'_, PyDict> {
+        self.into_py_dict(py)
+    }
 }
```

Comment:
This is breaking, since any implementer will now be forced to implement the "new" method, but since this is just a simple rename it is _probably_ fine to do so.

Reply:
I have added the changelog entry & migration docs